src/core/create-pacts.ts

```diff
@@ -1,14 +1,12 @@
 import {PactConfig, Provider, readPactsConfig} from './read-pacts-config';
-import * as fs from 'fs';
 import * as tsMorph from 'ts-morph';
 import {getInteractionFromTsNode, Interaction} from './interactions';
 import {glob} from 'glob';
+import {writeToFile} from '../utils/writeToFile';
+import {printInteraction} from './printInteraction';
 
 export function createPacts() {
     const pactsConfig = readPactsConfig();
-    if (!fs.existsSync(pactsConfig.buildDir)) {
-        fs.mkdirSync(pactsConfig.buildDir);
-    }
 
     pactsConfig.providers.forEach((provider) => createPactForProvider(provider, pactsConfig));
```

Comment:
You should separate generating pacts from saving it to file system. It would be more functional and much easier to test, example:
`const generatedPacts = pactsConfig.providers.map((provider) => createPactForProvider(provider, pactsConfig));`
`writeToFile(pactsConfig.buildDir, generatedPacts .resultFilePath, generatedPacts .resultJSON);`

and for example extract `pactsConfig.providers.map((provider) => createPactForProvider(provider, pactsConfig));` to separate function, import it in tests suite and much easier generate snapshots without mocking fs module.

Reply:
Done.

I've also moved read pacts config file outside of that function.